src/twisted/test/test_twistd.py

```diff
@@ -1542,9 +1542,10 @@ def test_getLogObserverFile(self):
         """
         logFiles = _patchTextFileLogObserver(self.patch)
         filename = self.mktemp()
-        logger = app.AppLogger({"logfile": filename})
+        sut = app.AppLogger({"logfile": filename})
 
-        logger._getLogObserver()
+        observer = sut._getLogObserver()
+        self.addCleanup(logger.globalLogPublisher.removeObserver, observer)
```

Comment:
this is the global logger and this test was adding an observer but not removing it at the end

the created observer was opening a temp file. which was not closed at the end

this resulted in random failures in other tests as the garbage collector was complaining about the file that was left open.